File: assessment.js
class Story {
    constructor( length, moralLesson, ageGroup) {
      this.length = length;
      this.moralLesson = moralLesson;
      this.ageGroup = ageGroup;
    }
  }
class Storyteller {
    constructor(name, stories) {
      this.name = name;
      this.stories = stories;
    }


tellStory(story) {
    print("this is a story of a dog and a cat")
  }
}
class Translator {
    constructor(sourceLanguage, targetLanguage) {
      this.sourceLanguage = sourceLanguage;
      this.targetLanguage = targetLanguage;
    }
}


class Legend extends Story {
    constructor(title, author, length, moralLesson, ageGroup) {
      super(title, author, length, moralLesson, ageGroup);
      this.author = author;
    }
  }

//   # 2. **African Cuisine:** You're creating a recipe app specifically for African cuisine.
//   # The app needs to handle recipes from different African countries, each with its
//   # unique ingredients, preparation time, cooking method, and nutritional
//   # information. Consider creating a `Recipe` class, and think about how you might
//   # create subclasses for different types of recipes (e.g., `MoroccanRecipe`,
//   # `EthiopianRecipe`, `NigerianRecipe`), each with their own unique properties and
//   # methods.


class Recipe {
    constructor( ingredients, preparationTime, cookingMethod, nutritionalInformation) {
    
      this.ingredients = ingredients;
      this.preparationTime = preparationTime;
      this.cookingMethod = cookingMethod;
      this.nutritionalInformation = nutritionalInformation;
    }
  
  
    get totalProtein() {
      return this.nutritionalInformation.protein;
    }
  
    
  
    printRecipe() {
      console.log(`
      Ingredients: ${this.ingredients.join(", ")}
      Preparation time: ${this.preparationTime} minutes
      Cooking method: ${this.cookingMethod}
      Nutritional information
        Protein: ${this.totalProtein}

      `);
    }
  }
  
  class MoroccanRecipe extends Recipe {
    constructor(name, ingredients, preparationTime, cookingMethod, nutritionalInformation, cuisine) {
      super( ingredients, preparationTime, cookingMethod, nutritionalInformation);
   
    }
  
    printRecipe() {
      super.printRecipe();
      console.log(`
      Cuisine: ${this.cuisine}
      `);
    }
  }
  
  class EthiopianRecipe extends Recipe {
    constructor(name, ingredients, preparationTime, cookingMethod, nutritionalInformation, cuisine) {
      super( ingredients, preparationTime, cookingMethod, nutritionalInformation);
      this.cuisine = cuisine;
    }
  
    printRecipe() {
      super.printRecipe();
      console.log(`
      Cuisine: ${this.cuisine}
      `);
    }
  }
  
  class NigerianRecipe extends Recipe {
    constructor( ingredients, preparationTime, cookingMethod, nutritionalInformation) {
      super( ingredients, preparationTime, cookingMethod, nutritionalInformation);
      
    }
  
    printRecipe() {
      super.printRecipe();
      console.log(`
      Cuisine: ${this.cuisine}
      `);
    }
  }
  
  const ingredients = ["flour", "yeast", "cooking oil"];
  const preparationTime = 3;
  const cookingMethod = "Baking";
  const nutritionalInformation = {
    calories: 500,
    fat: 20,
    protein: 4,
    carbohydrates: 5
  };
  
  const recipe = new MoroccanRecipe(
  
    ingredients,
    preparationTime,
    cookingMethod,
    nutritionalInformation,

  );
  
  recipe.printRecipe();



  class Product {
    constructor(fruit, price, quantity) {
      this.fruit = fruit;
      this.price = price;
      this.quantity = quantity;
    }
  
    get totalValue() {
      return this.price * this.quantity;
    }
  }
  
  const product1 = new Product(
    fruit= "Banana",
    price= 1,
    quantity= 1
  );
  
  const product2 = new Product(
    fruit = "Apple",
    price = 2,
    quantity= 2
  );
  
  const product3 = new Product(
    fruit= "Orange",
    price= 3,
    quantity=3
  );
  
  console.log(product1.totalValue); 
  console.log(product2.totalValue); 
  console.log(product3.totalValue);

File: assessment.py
# **Ancestral Stories:** In many African cultures, the art of storytelling is passed
# down from generation to generation. Imagine you're creating an application that
# records these oral stories and translates them into different languages. The
# stories vary by length, moral lessons, and the age group they are intended for.
# Think about how you would model `Story`, `StoryTeller`, and `Translator`
# objects, and how inheritance might come into play if there are different types of
# stories or storytellers.




class Story:
    def __init__(self, length, moral_lesson, age_group):
        self.length= length
        self.moral_lesson = moral_lesson
        self.age_group = age_group

    def get_length(self):
        return len(self.content)

    def display(self):
        print(f"this is book is called: {self.title}")
        print(f"and it mainly talks about: {self.content}")
        print(f"some of the lessons to learn from it includes: {self.moral_lesson}")
        print(f"it was written for people of age group ranged between: {self.age_group}")

class StoryTeller:
    def __init__(self, name, language):
        self.name = name
        self.language = language

    def tell_story(self, story):
        print(f"Storyteller: {self.name}")
        print(f"Language: {self.language}")
        story.display()


# 2. **African Cuisine:** You're creating a recipe app specifically for African cuisine.
# The app needs to handle recipes from different African countries, each with its
# unique ingredients, preparation time, cooking method, and nutritional
# information. Consider creating a `Recipe` class, and think about how you might
# create subclasses for different types of recipes (e.g., `MoroccanRecipe`,
# `EthiopianRecipe`, `NigerianRecipe`), each with their own unique properties and
# methods.


class Recipe:


  def __init__(self, name, ingredients, preparation_time, cooking_method, nutritional_information):
   

  
    self.name = name
    self.ingredients = ingredients
    self.preparation_time = preparation_time
    self.cooking_method = cooking_method
    self.nutritional_information = nutritional_information

  def __repr__(self):

    return f"Recipe({self.name}, {self.ingredients}, {self.preparation_time}, {self.cooking_method}, {self.nutritional_information})"

class MoroccanRecipe(Recipe):
 

  def __init__(self, name, ingredients, preparation_time, cooking_method, nutritional_information, cuisine):
   

 
    super().__init__(name, ingredients, preparation_time, cooking_method, nutritional_information)
    self.cuisine = cuisine

recipe = MoroccanRecipe(
  name="Moroccan Bread",
  ingredients=["flour", "yeast", "cooking oil"],
  preparation_time=3,
  cooking_method="Baking",
  nutritional_information={"Calories": 200, "Fat": 10, "Protein": 5, "Carbohydrates": 25},
  cuisine="Moroccan"
)

print(recipe)


class Product:
  

  def __init__(self, name, price, quantity):
 

    self.name = name
    self.price = price
    self.quantity = quantity

  def total_value(self):
    return self.price * self.quantity



product1 = Product(
  name="Banana",
  price=1,
  quantity=1,
)

product2 = Product(
  name="Apple",
  price=2,
  quantity=2,
)

product3 = Product(
  name="Orange",
  price=3,
  quantity=3,
)


print(product1.total_value())  
print(product2.total_value())  
print(product3.total_value())  
    

